cmd/src/server.rs

```diff
@@ -234,6 +215,47 @@ impl TiKVServer {
         ConfigController::new(config, version)
     }
 
+    fn register_config(
+        mut config: TiKvConfig,
+        pd_client: Arc<RpcClient>,
+    ) -> (TiKvConfig, configpb::Version) {
+        if config.server.advertise_addr.is_empty() {
+            info!(
+                "no advertise-addr is specified, falling back to default addr";
+                "addr" => %config.server.addr
+            );
+            config.server.advertise_addr = config.server.addr.clone();
+        }
+        // Advertise address and cluster id can not be changed
+        let advertise_addr = config.server.advertise_addr.clone();
+        let cluster_id = config.server.cluster_id;
+        // Using the same file for initialize global logger
+        // and diagnostics service
+        let log_file = config.log_file.clone();
+
+        match ConfigHandler::create(advertise_addr.clone(), pd_client, config.clone()) {
+            Ok((v, mut cfg)) => {
+                cfg.server.advertise_addr = advertise_addr;
+                cfg.server.cluster_id = cluster_id;
+                cfg.log_file = log_file;
+                cfg.enable_dynamic_config = true;
+                (cfg, v)
+            }
+            Err(e) => {
+                if let Some(PdError::Grpc(grpcio::Error::RpcFailure(status))) =
+                    e.downcast_ref::<PdError>()
```

Comment:
Using `downcast_ref` is not recommended, you can define CfgError

```rust
enum CfgError {
  Other(Box<dyn error::Error + Sync + Send>),
  Pd(PdError)
}
```
Then we can use pattern match to get the umimplemented error.